README.md
Name: Álvaro Maza Montesinos
PID: A17655072
Link: https://elaborate-heliotrope-bcb372.netlify.app/


changelog.md
Added on HW4

- Bussiness analyticts section below about me section using grid.

- Third party JavaScript to embed an instagram post of the business analytics association.

- Created SVG logo that links to the association webpage and styled it as a button.

- Try to change other icons to SVG, but they lost quality, so I left them as png.

- Eliminated navigation bar for small devices since it did not work well on them, would have like to develop 
a hamburguer menu but i did not have enough time.

- Change about me flex display to behave as a block in small devices to increase readability.

blog.js
var app = new function() {

  this.el = document.getElementById('posts');

  if (!localStorage.getItem('postsdata')) {
    const myPosts = [['Example 1', '2023-03-14','I hope the code is not awful'],['Example 2', '2023-03-14','Have a good day :)']];
    localStorage.setItem('postsdata', JSON.stringify(myPosts));
  }


  this.posts = JSON.parse(localStorage.getItem('postsdata'));
  
  this.FetchAll = function() {
    var data = '';
    localStorage.setItem('postsdata', JSON.stringify(this.posts));

    if (this.posts.length > 0) {
      for (var i = 0; i < this.posts.length; i++) {
        data += '<tr>';
        data += `<td> <b>Title: </b> ${this.posts[i][0]} <b>Date: </b> ${this.posts[i][1]} <b>Summary: </b> ${this.posts[i][2]}</td>`;
        data += '<td><button onclick="app.Edit(' + i + ')">Edit</button></td>';
        data += '<td><button onclick="app.Delete(' + i + ')">Delete</button></td>';
        data += '</tr>';
      }
    }
    return this.el.innerHTML = data;
  };

  this.Add = function () {
    var el1 = document.getElementById('add-title');
    var el2 = document.getElementById('add-date');
    var el3 = document.getElementById('add-summary');
    // Get the value
    var post = [DOMPurify.sanitize(el1.value),DOMPurify.sanitize(el2.value),DOMPurify.sanitize(el3.value)];

    if (post) {
      // Add the new value
      this.posts.push(post);
      // Reset input value
      el1.value = '';
      el2.value = '';
      el3.value = '';
      // Dislay the new list
      this.FetchAll();
    }
    document.getElementById('add').close();
  };

  this.Edit = function (item) {
    var el1 = document.getElementById('edit-title');
    var el2 = document.getElementById('edit-date');
    var el3 = document.getElementById('edit-summary');
    // Display value in the field
    el1.value = this.posts[item][0];
    el2.value = this.posts[item][1];
    el3.value = this.posts[item][2];
    // Display fields
    document.getElementById('edit').show();
    self = this;

    document.getElementById('saveEdit').onsubmit = function() {
      // Get value
      var post = [DOMPurify.sanitize(el1.value),DOMPurify.sanitize(el2.value),DOMPurify.sanitize(el3.value)];

      if (post) {
        // Edit value
        self.posts.splice(item, 1, post);
        // Display the new list
        self.FetchAll();
        // Hide fields
        CloseInput();
      }
    }
  };

  this.Delete = function (item) {
    // Delete the current row
    this.posts.splice(item, 1);
    // Display the new list
    this.FetchAll();
  };
  
}


function CloseInput() {
  document.getElementById('edit').close();
}

function OpenApp(){
  document.getElementById('add').show();
}

function CloseApp(){
  document.getElementById('add').close();
}


export {app,CloseInput,OpenApp,CloseApp};


customdialog.js
var output = document.getElementById('result');
var dialog = document.getElementById("alert_dialog");
var dialog2 = document.getElementById("confirm_dialog");
var dialog3 = document.getElementById("prompt_dialog");



document.getElementById('alert').addEventListener('click', () => {
    setTimeout(() => {
        dialog.show();
    },0);
});

document.getElementById('ok').addEventListener('click', () => {
    setTimeout(() => {
        dialog.close();
    },0);
});

document.getElementById('confirm').addEventListener('click', () => {
    output.innerHTML = '';
    setTimeout(() => {
        dialog2.show();
    },0);
});

document.getElementById('conf_ok').addEventListener('click', () => {
    setTimeout(() => {
        output.innerHTML = 'The value returned by the confirm method is : True';
        dialog2.close();
    },0);
});

document.getElementById('conf_cancel').addEventListener('click', () => {
    setTimeout(() => {
        output.innerHTML = 'The value returned by the confirm method is : False';
        dialog2.close();
    },0);
});



document.getElementById('safe_prompt').addEventListener('click', () => {
    output.innerHTML = '';
    setTimeout(() => {
        dialog3.show();
    },0);
});

document.getElementById('prompt_ok').addEventListener('click', () => {
    setTimeout(() => {
        let safe_input = DOMPurify.sanitize(document.getElementById('prompt_input').value);
        if (safe_input!= ''){
        output.innerHTML = `Prompt result: ${safe_input}`;
        }
        else{
            output.innerHTML = `User didn’t enter anything`;
        };
        dialog3.close();
    },0);
});

document.getElementById('prompt_cancel').addEventListener('click', () => {
    setTimeout(() => {
        output.innerHTML = `User didn’t enter anything`;
        dialog3.close();
    },0);
});




styledblog.js
var app = new function() {

    this.el = document.getElementById('posts');
  
    if (!localStorage.getItem('postsdata')) {
      const myPosts = [['Example 1', '2023-03-24','I hope the code is not awful'],['Example 2', '2023-03-14','Have a good day :)']];
      localStorage.setItem('postsdata', JSON.stringify(myPosts));
    }
    this.posts = JSON.parse(localStorage.getItem('postsdata'));
    
    this.FetchAll = function() {
      var data = '';
      localStorage.setItem('postsdata', JSON.stringify(this.posts));
  
      if (this.posts.length > 0) {
        for (var i = 0; i < this.posts.length; i++) {
          data += '<tr>';
          data += `<td> <b>Title: </b> ${this.posts[i][0]} <b>Date: </b> ${this.posts[i][1]} <b>Summary: </b> ${this.posts[i][2]}</td>`;
          data += '<td><img class="logo" src="images/pencil.png" onclick="app.Edit(' + i + ')"></img></td>';
          data += '<td><img class="logo" src="images/trash.png" onclick="app.Delete(' + i + ')"></img></td>';
          data += '</tr>';
        }
      }
      return this.el.innerHTML = data;
    };
  
    this.Add = function () {
      var el1 = document.getElementById('add-title');
      var el2 = document.getElementById('add-date');
      var el3 = document.getElementById('add-summary');

      var post = [DOMPurify.sanitize(el1.value),DOMPurify.sanitize(el2.value),DOMPurify.sanitize(el3.value)];
  
      if (post) {
        this.posts.push(post);

        el1.value = '';
        el2.value = '';
        el3.value = '';

        this.FetchAll();
      }
      document.getElementById('add').close();
    };
  
    this.Edit = function (item) {
      var el1 = document.getElementById('edit-title');
      var el2 = document.getElementById('edit-date');
      var el3 = document.getElementById('edit-summary');

      el1.value = this.posts[item][0];
      el2.value = this.posts[item][1];
      el3.value = this.posts[item][2];

      document.getElementById('edit').show();
      self = this;
  
      document.getElementById('saveEdit').onsubmit = function() {
        var post = [DOMPurify.sanitize(el1.value),DOMPurify.sanitize(el2.value),DOMPurify.sanitize(el3.value)];
        if (post) {
          self.posts.splice(item, 1, post);
          self.FetchAll();
          CloseInput();
        }
      }
    };
  
    this.Delete = function (item) {
      this.posts.splice(item, 1);
      this.FetchAll();
    };
    
}
  
function CloseInput() {
    document.getElementById('edit').close();
}

function OpenApp(){
    document.getElementById('add').show();
}

function CloseApp(){
    document.getElementById('add').close();
}

export {app,CloseInput,OpenApp,CloseApp};